### PR TITLE
Prevent model.Writer from inserting empty text nodes

### LINKED ISSUE #1320: model.Writer should not be letting creating empty text nodes
At this moment it is possible to create an empty text node (`.data == ''`) using `model.Writer` methods. Such text node is incorrect. Because it stays in model tree, it causes bugs.

`model.Writer` should throw when trying to create an empty text node.

### PR DESCRIPTION
**Pull request by scofalik**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Fix: Prevent `model.Writer` from inserting empty text nodes. Closes #1320.

---

### Additional information

Alternative solution to: https://github.com/ckeditor/ckeditor5-engine/pull/1321

### CODE REVIEW

File: src/model/writer.js [hunk 1]
```diff
@@ -222,6 +226,10 @@ export default class Writer {
 	 * third parameter is a {@link module:engine/model/item~Item model item}.
 	 */
 	insertText( text, attributes, itemOrPosition, offset ) {
+		if ( text == '' ) {
```
Comment (Reinmar) [inline src/model/writer.js]: Is this necessary?

**Comment by Reinmar** [comment]
I added tests for inserting an array of items and those empty text nodes there are normalized. That's weird that inserting a single text node fails.

**Comment by Reinmar** [comment]
I think that the patch should be somewhere in text nodes normalization.

**Comment by scofalik** [comment]
This PR is a bit outdated - I'll create a new one.

**Pull request closed without merging**